src/settingsmanager.cpp

```diff
@@ -7,6 +7,50 @@
 #include <QLocale>
 #include <QList>
 
+/*
+ * Returns the default directory (UTF-8 encoded) where downloaded files should be saved.
+ * Can be overriden via the options parameter of `kiwix::Downloader::startDownload()`. 
+ * Its path can vary and is determined as follows:
+ *
+ * On Windows:
+ *   * `$APPDATA/kiwix` if environment variable `$APPDATA` set, *otherwise...*
+ *   * `$USERPROFILE/kiwix` if environment variable `$USERPROFILE` set, *otherwise...*
+ * On other OSs:
+ *   * `$XDG_DATA_HOME/kiwix` if environment variable `$XDG_DATA_HOME` set, *otherwise...*
+ *   * `$HOME/.local/share/kiwx` if environment variable `$HOME` set, *otherwise...*
+ *
+ * * Current working directory.
+ */
+std::string getDataDirectory()
+{
+  std::string dataDir;
+#ifdef _WIN32
+  wchar_t* cDataDir = ::_wgetenv(L"APPDATA");
+  if (cDataDir == nullptr)
+    cDataDir = ::_wgetenv(L"USERPROFILE");
+  if (cDataDir != nullptr)
+    dataDir = WideToUtf8(cDataDir);
+#else
+  char* cDataDir = ::getenv("XDG_DATA_HOME");
+  if (cDataDir != nullptr) {
+    dataDir = cDataDir;
+  } else {
+    cDataDir = ::getenv("HOME");
+    if (cDataDir != nullptr) {
+      dataDir = cDataDir;
+      dataDir = kiwix::appendToDirectory(dataDir, ".local");
+      dataDir = kiwix::appendToDirectory(dataDir, "share");
+    }
+  }
+#endif
```

Comment:
I wonder if we could remove this and replace it with Qt function (writableLocation)[`https://doc.qt.io/qt-6/qstandardpaths.html#writableLocation`] (or (standardLocation)[https://doc.qt.io/qt-6/qstandardpaths.html#standardLocations], depending if we want to read data (configuration, zim files) or write it)

`QStandardPaths::AppDataLocation` seems to return what the commented lines are doing (at least on Windows and Linux)

Reply:
Good idea. I looked into it and the only issue I see is the intermediate differences between Qt and us currently For example, I have the resolutions below: 

## Windows
Current:

1. `C:\Users\<USER>\AppData\Roaming\kiwix`
2. `C:\Users\<USER>\kiwix`
3. `<APPDIR>`

AppDataLocation:

1. `C:/Users/<USER>/AppData/Roaming/kiwix`
2. `C:/ProgramData/kiwix`
4. `<APPDIR>`
5. `<APPDIR>/data`
6. `<APPDIR>/data/kiwix`

## Rest
Current:

1. `~/.local/share/kiwix`
2. `<APPDIR>`

AppDataLocation:

1. `~/.local/share/<APPNAME>`
3. `/usr/local/share/<APPNAME>`
4. `/usr/share/<APPNAME>`

How should we go about this?